executors/icu4j/74/executor-icu4j/src/main/java/org/unicode/conformance/testtype/datetimeformatter/DateTimeFormatterTester.java

```diff
@@ -0,0 +1,169 @@
+package org.unicode.conformance.testtype.datetimeformatter;
+
+import java.text.ParseException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+
+import java.util.Date;
+
+import com.ibm.icu.util.Calendar;
+import com.ibm.icu.text.DateFormat;
+import com.ibm.icu.util.ULocale;
+
+import io.lacuna.bifurcan.IMap;
+import io.lacuna.bifurcan.Map;
+
+
+import org.unicode.conformance.ExecutorUtils;
+import org.unicode.conformance.testtype.ITestType;
+import org.unicode.conformance.testtype.ITestTypeInputJson;
+import org.unicode.conformance.testtype.ITestTypeOutputJson;
+
+public class DateTimeFormatterTester implements ITestType {
+
+  public static DateTimeFormatterTester INSTANCE = new DateTimeFormatterTester();
+
+  @Override
+  public ITestTypeInputJson inputMapToJson(Map<String, Object> inputMapData) {
+    DateTimeFormatterInputJson result = new DateTimeFormatterInputJson();
+
+    result.label = (String) inputMapData.get("label", null);
+    result.locale = (String) inputMapData.get("locale", null);
+    result.skeleton = (String) inputMapData.get("skeleton", null);
+
+    result.inputString = (String) inputMapData.get("input_string", null);
+
+    java.util.Map<String, Object> inputOptions =
+        (java.util.Map<String, Object>) inputMapData.get("options", null);
+
+    result.timeZone = (String) inputOptions.get("timeZone");
+    ZoneId thisZoneId;
+    if (result.timeZone == null) {
+      thisZoneId = ZoneId.systemDefault();
+    } else {
+      thisZoneId = ZoneId.of(result.timeZone);
+    }
+
+    String inputStringDateTime = result.inputString.substring(0, 25);
+    int indexLeftBracket = result.inputString.indexOf("[");
+    int indexRightBracket = result.inputString.indexOf("]", indexLeftBracket);
+    String inputTimeZone = result.inputString.substring(indexLeftBracket, indexRightBracket);
```

Comment:
This is unused. Take a failing test, add as unit test (temporarily), see what happens with this code & whether you can parse the time zone from this code vs. the `thisZoneId` code. If this code is unhelpful, remove it.